packages/block-editor/src/components/block-preview/index.js

```diff
@@ -33,19 +37,57 @@ function BlockPreview( props ) {
 }
 
 export function BlockPreviewContent( { blocks, settings } ) {
+	const previewRef = useRef( null );
+
+	const [ previewScale, setPreviewScale ] = useState( 1 );
+	const [ visibility, setVisibility ] = useState( 'hidden' );
+
+	let timerId;
+
+	// Dynamically calculate the scale factor
+	useLayoutEffect( () => {
+		const { clientId } = blocks;
+		timerId = setTimeout( () => {
+			const refNode = previewRef.current;
+			const containerWidth = refNode.offsetWidth - ( 14 * 2 );
+
+			const previewDomElements = getClockPreviewContainerDOMNode( clientId );
```

Comment:
I imagine that if we render multiple previews of the same block (same id), this could be ambiguious, should we add a ref to a parent to limit the scope of the selector?

Reply:
I've fixed this by adding a `scope` param to the DOM util and passing a relevant scope into the call to get the Block contents.